src/auxiliary/p4est.jl

```diff
@@ -24,6 +24,18 @@ function init_p4est()
     return nothing
 end
 
+# for convenience to either pass a Ptr or a PointerWrapper
+const P4estPtrOrPW = Union{Ptr{p4est_t}, PointerWrapper{p4est_t}}
+const P8estPtrOrPW = Union{Ptr{p8est_t}, PointerWrapper{p8est_t}}
+
+const P4estConnectvitytPtrOrPW = Union{Ptr{p4est_connectivity_t},
+                                       PointerWrapper{p4est_connectivity_t}}
+const P8estConnectvitytPtrOrPW = Union{Ptr{p8est_connectivity_t},
+                                       PointerWrapper{p8est_connectivity_t}}
+
+const P4estGhostPtrOrPW = Union{Ptr{p4est_ghost_t}, PointerWrapper{p4est_ghost_t}}
+const P8estGhostPtrOrPW = Union{Ptr{p8est_ghost_t}, PointerWrapper{p8est_ghost_t}}
```

Comment:
I am wondering if defining something like
```julia
const PointerOrWrapper = Union{Ptr{T}, PointerWrapper{T}} where {T}
```
and then using it as
```julia
fun(arg::PointerOrWrapper{p8est_ghost_t})
```
in the code would be more descriptive? It's just a suggestion, I am myself not sure which way is better - I just thought it might scale better and not introduce more cryptic names.

Reply:
I like the idea and implemented it in [4cf01a5](https://github.com/trixi-framework/Trixi.jl/pull/1434/commits/4cf01a512988f9620e064f033c84690e1f40c786).